### PR TITLE
added flyway script, model, repository and tests

### PR DESCRIPTION
**Pull request by NikoIskra**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -39,6 +43,14 @@ dependencies {
     // https://mvnrepository.com/artifact/javax.annotation/javax.annotation-api
     implementation 'javax.annotation:javax.annotation-api:1.3.2'
 
+    implementation 'jakarta.validation:jakarta.validation-api:3.0.2'
+
+    testImplementation 'org.testcontainers:junit-jupiter:1.19.0'
+
+    // https://mvnrepository.com/artifact/com.h2database/h2
+    testImplementation 'com.h2database:h2:2.2.222'
```
Comment (vitalii-martynovskyi-gl) [inline app/build.gradle:51]: I see both `testcontainers` and `h2` please keep only  that which you use

File: app/src/main/java/com/account/persistence/entity/Account.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name= "account")
+@Data
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/java/com/account/persistence/entity/Account.java:19]: This entity already has getters and setters and constructor. At the same time, you put Lombok annotations, which looks odd.
However, I would not recommend using Lombok annotations with entities. This may have an influence over  EntityManager
So, let's write without Lombok annotations on entities.

File: app/src/main/java/com/account/persistence/entity/Account.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    
+    private String email;
+
+    private String username;
+
+    private String password;
+
+    private String status;
+
+    private LocalDateTime create_at;
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/java/com/account/persistence/entity/Account.java:36]: Please, follow the Java naming convention.
Make this field not insertable, check `@Column` annotation
Use Long type for this date

File: app/src/main/resources/db/migration/V1_0_create_account_schema.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE IF NOT EXISTS account
+(
+    id uuid,
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_schema.sql:3]: Do you decide to generate an `id` on the backend?

File: app/src/main/resources/db/migration/V1_0_create_account_role.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE IF NOT EXISTS account_role
+(
+    id uuid,
+    account_id uuid,
+    role character varying(8) NOT NULL,
+    status character varying(8) NOT NULL,
+    create_at timestamp without time zone NOT NULL DEFAULT now(),
+    PRIMARY KEY (id),
+    UNIQUE (account_id),
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_role.sql:9]: this will not allow to have more the one role for each account, which makes this table useless, check ticket pls

File: app/src/main/resources/db/migration/V1_0_create_account_schema.sql [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    email character varying(128) NOT NULL,
+    username character varying(64) NOT NULL,
+    password character varying(128) NOT NULL,
+    status character varying(8) NOT NULL,
+    create_at timestamp without time zone DEFAULT now(),
+    update_at timestamp without time zone,
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_schema.sql:9]: please rename all occurrences of `create_at -> created_at` and `update_at -> updated_at`
my mistake in the ticket(fixed)

File: app/src/main/resources/db/migration/db_schema.sql [hunk 7]
```diff
@@ -0,0 +1 @@
+CREATE SCHEMA IF NOT EXISTS account_service;
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/db_schema.sql:1]: You don't need this.
just separate DBs per service  `flyway.url = jdbc:postgresql://localhost:5432/account_service`

File: template-openAPI.json [hunk 8]
```diff
@@ -14,6 +14,45 @@
             }
           }
         }
+      },
+      "/api/v1/account": {
```
Comment (vitalii-martynovskyi-gl) [inline template-openAPI.json:18]: not in this PR

File: app/src/main/java/com/account/persistence/entity/Account.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+
+    public Optional<AccountDTO> map(Object object) {
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/java/com/account/persistence/entity/Account.java:85]: This method shouldn't be here

File: app/src/test/java/com/account/test/JPAAccountTest.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  @Test
+  public void insertToDB() {
+    Account account = new Account("testmail", "testusername", "testpassword");
+    accountRepository.save(account);
+    List<Account> accounts = accountRepository.findAll();
+    assertEquals(1, accounts.size());
+  }
+
```
Comment (vitalii-martynovskyi-gl) [inline app/src/test/java/com/account/test/JPAAccountTest.java:38]: add tests for negative scenarios like missed mandatory data 
also, add the assertion that values persisted correctly

File: app/src/test/java/com/account/test/JPAAccountTest.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  @Test
+  public void expectEmptyList() {
+    List<Account> accounts = accountRepository.findAll();
+    assertEquals(0, accounts.size());
+  }
+
+  @Test
+  public void insertToDB() {
+    Account account = new Account("testmail", "testusername", "testpassword");
+    accountRepository.save(account);
```
Comment (vitalii-martynovskyi-gl) [inline app/src/test/java/com/account/test/JPAAccountTest.java:34]: Seems like this test is not working. Because, status field  has "NOT NULL" constraint

File: app/src/main/resources/db/migration/V1_0_create_account_schema.sql
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_schema.sql:1]: The file name for migration scripts should start from the incremental versions, and then some descriptions 
`V1__create_account_schema.sql`
See:
https://github.com/flyway/flywaydb.org/blob/gh-pages/documentation/concepts/migrations.md#naming

File: app/src/main/resources/db/migration/V1_0_create_account_role.sql
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_role.sql:1]: `V2__create_account_role.sql`

File: app/src/main/resources/db/migration/V1_0_create_account_schema.sql [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+    PRIMARY KEY (id),
+    UNIQUE (email, username)
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/resources/db/migration/V1_0_create_account_schema.sql:11]: do not combine this, check ticket

File: app/src/main/java/com/account/persistence/entity/Account.java [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+    private String username;
+
+    private String password;
+
+    private String status;
+
+    private LocalDateTime create_at;
+
+
```
Comment (vitalii-martynovskyi-gl) [inline app/src/main/java/com/account/persistence/entity/Account.java:38]: Missed updated_at
`updated_at`, timestamp(no time zone), configure entity to update on every write operation

**Comment by vitalii-martynovskyi-gl** [comment]
Other thing is branches and commits @NikoIskra 
### Commits
because of this PR related to issue/ticket #1, you need to reference on it in the commit messages, like: `#1 impl flyway scripts`,  `#1 create entity` ... 

### Branches
When you are working on some feature/issue/ticket there is such a concept `feature branch`
https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow
check this and follow it in the next PRs, this one keep as it is.
the branch names also  should have the reference on the ticket like `feature/#1_account_db `
